src/debug_kernel.h

```diff
@@ -0,0 +1,36 @@
+/*
+Copyright (C) 2015 The University of Notre Dame
+This software is distributed under the GNU General Public License.
+See the file LICENSE for details.
+*/
+
+#ifndef DEBUG_KERNEL_H
+#define DEBUG_KERNEL_H
+
+#include "kerneltypes.h"
+
+// This is a set of functions we can call in kernel mode to facilitate
+// debugging in kernel mode. To enable them, uncomment the following definition:
+
+enum register_name_t {
+    reg_esi = 0,
+    reg_edi,
+    reg_ebp,
+    reg_esp
+};
+
+/**
+ * @brief   Dump memory content of a given length at a given address
+ * @detail  Dump memory content in hex, starting or ending at a given address
+ *          depending on the given length.
+ *
+ * @param   addr    The start or end address of memory to be dumped
+ * @param   len     The length of memory to be dumped in 4 bytes (32-bit
```

Comment:
Is there a case where we need the negative length functionality? Seems like it could lead to strange issues since the behavior is not that intuitive.